src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/Conversions.cs

```diff
@@ -287,46 +287,36 @@ private static Conversion ToConversion(OverloadResolutionResult<MethodSymbol> re
 
         public override Conversion GetStackAllocConversion(BoundStackAllocArrayCreation sourceExpression, TypeSymbol destination, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
-            Debug.Assert((object)sourceExpression.Type == null);
-            Debug.Assert(sourceExpression.ElementType != null);
-
-            var pointerConversion = default(Conversion);
-
             if (sourceExpression.Syntax.IsVariableDeclarationInitialization())
```

Comment:
>IsVariableDeclarationInitialization [](start = 40, length = 35)

Since you moved `IsLegalSpanStackAllocPosition()` to binding, `IsVariableDeclarationInitialization()` will always be true here, right?

Reply:
Also, does this mean this will reject ternary operators?

---
In reply to: [139543076](https://github.com/dotnet/roslyn/pull/22162#discussion_r139543076) [](ancestors = 139543076)

Reply:
It will not be true when it is not in an initializer. Then it will already have a span type, so we do not need to look for special conversions.
If it is directly in an initializer, then it would have not type since it still has a chance to become pointer or span through stackalloc conversions.

---
In reply to: [139543224](https://github.com/dotnet/roslyn/pull/22162#discussion_r139543224) [](ancestors = 139543224,139543076)